sql/core/src/test/scala/org/apache/spark/sql/CachedTableSuite.scala

```diff
@@ -83,25 +83,6 @@ class CachedTableSuite extends QueryTest with SQLTestUtils with SharedSQLContext
     }.sum
   }
 
-  test("withColumn doesn't invalidate cached dataframe") {
-    var evalCount = 0
-    val myUDF = udf((x: String) => { evalCount += 1; "result" })
-    val df = Seq(("test", 1)).toDF("s", "i").select(myUDF($"s"))
-    df.cache()
```

Comment:
Here is when evalCount is incremented, the rest of the tests have no effect on the value of evalCount.

`evalCount` is copied (by value) to the closure and therefore it's not possible to change the value `evalCount` defined in the driver by incrementing it in the UDF (because each closure has a different copy of `evalCount`)